src/main/java/net/syscon/elite/security/AuthenticationFacade.java

```diff
@@ -44,14 +44,15 @@ public String getCurrentUsername() {
         return username;
     }
 
-    public boolean isIdentifiedAuthentication() {
+    public AuthSource getProxyUserAuthenticationSource() {
         final var auth = getAuthentication();
-        return StringUtils.isNotBlank(MDC.get(PROXY_USER))
-                && Optional.ofNullable(auth).
+        return Optional.ofNullable(auth).
+                filter(a -> StringUtils.isNotBlank(MDC.get(PROXY_USER))).
                 filter(AuthAwareAuthenticationToken.class::isInstance).
                 map(AuthAwareAuthenticationToken.class::cast).
                 filter(AbstractAuthenticationToken::isAuthenticated).
-                map(AuthAwareAuthenticationToken::isNomisSource).orElse(Boolean.FALSE);
+                map(AuthAwareAuthenticationToken::getAuthSource).
+                orElse(AuthSource.NONE);
```

Comment:
Not really an issue.  But prefer the supplier (lazy) version of this.  `.orElseGet(() -> ...);`.  I have been burnt in the past with one.